Framework/Beamline/src/DetectorInfo.cpp

```diff
@@ -398,5 +411,17 @@ DetectorInfo::buildMergeIndices(const DetectorInfo &other) const {
   return merge;
 }
 
+void DetectorInfo::checkIdentitcalIntervals(const DetectorInfo &other,
+                                            const size_t linearIndex1,
+                                            const size_t linearIndex2) const {
```

Comment:
With the code extracted to a function it is not clear any more which index (1 or 2) corresponds to `this` and which to `other`. Maybe rename the indices?